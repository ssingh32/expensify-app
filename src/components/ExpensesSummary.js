import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal, nonVisibleExpenses}) => {
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const nonVisibleExpenseWord = nonVisibleExpenses <= 1 ? 'expense': 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,00.00');
   return (
    <div className={"page-header"}>
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
            {nonVisibleExpenses > 0 ? 
                <p className="page-header__title">Not showing <span>{nonVisibleExpenses}</span> {nonVisibleExpenseWord} because of filters</p> 
                :
                <p className="page-header__title">Showing all {expenseWord} no filters in place</p>
            }
            <div className="page-header__actions">
                <Link className="button" to="create">Add Expense</Link>
            </div>
        </div>
    </div>
   );
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount : visibleExpenses.length,
        expensesTotal : selectExpensesTotal(visibleExpenses),
        nonVisibleExpenses: state.expenses.length - visibleExpenses.length
    };
};

export default connect(mapStateToProps)(ExpensesSummary);