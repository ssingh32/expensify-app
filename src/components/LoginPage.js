import React from 'react';
import {connect} from 'react-redux';
import { startLogin, startGitHubLogin } from '../actions/auth';

export const LoginPage = ({ startLogin, startGitHubLogin }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1>Expensify App</h1>
                <p>It's time to get your expenses under control.</p>
                <div className="box-layout__button">
                    <button className="button" onClick={startLogin}>Login With Google</button>
                    <button className="button" onClick={startGitHubLogin}>Login With Github</button>
                </div>
            </div>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startGitHubLogin: () => dispatch(startGitHubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);