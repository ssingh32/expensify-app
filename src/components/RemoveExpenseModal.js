import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (
  <Modal
  isOpen={props.modalOpen}
  onRequestClose={props.clearSelectedOption}
  contentLabel="Remove Expense"
  className="modal"
  closeTimeoutMS={200}
  className="modal"
  >
  <h3 className="modal__title">Are You Sure?</h3>
  <div className="modal__button">
    <button className="button" onClick={props.onRemove}>Yes</button>
    <button className="button" onClick={props.handleModalOption}>No</button>
  </div>
  </Modal>
);

export default RemoveExpenseModal;