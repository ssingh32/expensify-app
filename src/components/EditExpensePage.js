import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveExpenseModal from '../components/RemoveExpenseModal';

export class EditExpensePage extends React.Component {  
  state = {
    modalOpen: false
  };

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }

  handleModalOption = () => {
      this.setState(() => ({modalOpen: !this.state.modalOpen}))
  }

  clearSelectedOption = () => {
    this.setState(() => ({modalOpen: false}))
  }

  render() {
    return (
    <div>
      <div className="page-header">
        <div className="content-container">
          <h1 className="page-header__title">Edit Expense</h1>
        </div>
      </div>
      <div className="content-container">
        <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.onSubmit}
        />
        <button className="button button--secondary" onClick={this.handleModalOption}>Remove Expense</button>
        <RemoveExpenseModal 
        handleModalOption={this.handleModalOption} 
        clearSelectedOption={this.clearSelectedOption} 
        onRemove={this.onRemove} 
        modalOpen={this.state.modalOpen}
        />
      </div>
    </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    startEditExpense :  (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense : (data) => dispatch(startRemoveExpense(data))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
