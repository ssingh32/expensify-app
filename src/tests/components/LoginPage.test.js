import React from 'react';
import {LoginPage} from '../../components/LoginPage';
import { shallow } from 'enzyme';


test('should render Login Page correctly', () => {   
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click', () => {
    let startLogin = jest.fn();
    let startGitHubLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} startGitHubLogin={startGitHubLogin} />);
    wrapper.find('button').at(0).simulate('click');
    expect(startLogin).toHaveBeenCalled();
})

test('Should call startGitHubLogin on button click', () => {
    let startLogin = jest.fn();
    let startGitHubLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} startGitHubLogin={startGitHubLogin} />);
    wrapper.find('button').at(1).simulate('click');
    expect(startGitHubLogin).toHaveBeenCalled();
})