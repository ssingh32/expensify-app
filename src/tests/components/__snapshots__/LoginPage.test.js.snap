// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1>
      Expensify App
    </h1>
    <p>
      It's time to get your expenses under control.
    </p>
    <div
      className="box-layout__button"
    >
      <button
        className="button"
      >
        Login With Google
      </button>
      <button
        className="button"
      >
        Login With Github
      </button>
    </div>
  </div>
</div>
`;
